### ISSUE TITLE
New check: `break` at the top of `while True` loop

### ISSUE THREAD

**Issue by nickdrozd** [PyCQA/pylint#8015]
### Current problem

A not-uncommon pattern is to check some condition at the top of a `while True` loop and break if the condition is met:

```python
while True:
    if condition:
        break

    do_stuff()
```

There is an instance of this in the Pylint codebase.

### Desired solution

It would be more straightforward and less verbose to negate the condition and move it into the `while` expression:

```python
while not condition:
    do_stuff()
```

Add a new check for the following circumstance:

1. There is a `while` loop with `True` condition (or more generally, constant condition);
2. the first statement of the loop body is an `if` statement; and
3. the first statement of the `if` body is a `break` statement.

In this case, emit a suggestion to negate the condition and move it to the `while`.

I don't know what the name of the warning should be. Suggestions welcome!

**Comment by yushao2** [comment]
it's finally done! thanks for this suggestion :)
